### PR TITLE
Add hostname expression

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

Add host/domain name expression in `on connect` event.

---
**Target Minecraft Versions:**  All
**Requirements:** 
**Related Issues:**  #3854

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("Hostname")
+@Description("The hostname/domain of the connecting player in a <a href='events.html#connect'>connect</a> event.")
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:38]:
```suggestion
@Description("The hostname used by the connecting player to connect to the server in a <a href='events.html#connect'>connect</a> event.")
```

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	public Class<String> getReturnType() {
+		return String.class;
+	}	
+	
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return "hostname";
+	}
+	
+}
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:81]:
```suggestion
}

```

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		"\tsend \"Welcome back tester!\""})
+@Since("INSERT VERSION")
+public class ExprHostname extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprHostname.class, String.class, ExpressionType.SIMPLE, "[the] (host|domain)[ ][name]");
+	}
+
+	@SuppressWarnings({"null", "unchecked"})
+	@Override
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:51]: just styling
```suggestion
	@Override
	@SuppressWarnings({"null", "unchecked"})
```

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(PlayerLoginEvent.class)) {
+			Skript.error("You can only use hostname expression in connect event.");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:54]:
```suggestion
			Skript.error("The hostname expression must only be used in a player connect event.");
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
just some minor suggestions ;D

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java (hunk 4 above)
Comment (TPGamesNL) [reply src/main/java/ch/njol/skript/expressions/ExprHostname.java:54]: I feel like it should be either 'can only' or 'must', not 'must only'

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+	static {
+		Skript.registerExpression(ExprHostname.class, String.class, ExpressionType.SIMPLE, "[the] (host|domain)[ ][name]");
+	}
+
+	@Override
+	@SuppressWarnings({"null", "unchecked"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerLoginEvent.class)) {
+			Skript.error("The hostname expression must be used in a player connect event.");
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:54]:
```suggestion
			Skript.error("The hostname expression must be used in a player connect event");
```

File: src/main/java/ch/njol/skript/expressions/ExprHostname.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+			return false;
+		}
+		return true;
+	}
+	
+	@Nullable
+	@Override
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprHostname.java:61]:
```suggestion
	@Override
	@Nullable
```

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**